types/index.ts

```diff
@@ -21,55 +7,17 @@ export type PageFeatureResult = {
   value: PageFeatureValue;
 };
 
-export interface PageFeatureGetter {
-  name: string;
-  func: () => Promise<PageFeatureResult>;
-}
-
-export type PageFeature =
-  | {
-      name: string;
-      error: true;
-    }
-  | {
-      name: string;
-      error: false;
-      version: number;
-      value: PageFeatureValue;
-    };
-
 export interface PageView {
   ts: number;
-  target?: boolean;
-  keyHash?: string;
   features: Record<string, PageFeatureResult>;
 }
 
-// Audiences
-
-export interface MatchedAudience {
-  id: string;
-  matchedAt: number;
-  expiresAt: number;
-  matchedOnCurrentPageView: boolean;
-}
-
-export type AudienceState = 'live' | 'paused' | 'deleted';
-
-export interface CachedAudienceMetaData {
-  cachedAt: number;
-  audiences: AudienceMetaData[];
-}
-
-export interface AudienceMetaData {
-  id: string;
-  version: number;
-}
+// Audience definition interfaces
 
-export type StringArrayQueryValue = string[];
+export type StringArrayQueryValue = Extract<PageFeatureValue, string[]>;
 
 export type VectorQueryValue = {
-  vector: number[];
+  vector: Extract<PageFeatureValue, number[]>;
```

Comment:
This can only be an array of floats - why do we need to Extract?

Reply:
The `Extract` evokes the semantic of dependency.
`vector` is a value used to match against values on `PageFeatureValue`, and for the proper working of the methods using then, they both have to be matching.
With `Extract`, if we change `PageFeatureValue` in the future, the compiler will tell us that `VectorQueryValue` should change too, or at least remember us that some other types depend on the values we changed and need to be revised.